### ISSUE TITLE
prediction is influenced by extra information in MISC column

### ISSUE THREAD

**Issue by jowagner** [CoNLL-UD-2018/UDPipe-Future#19]
To provide a baseline for PR #18, I trained a model on a treebank with no extra information in the `MISC` column and udpipe-future from my `lemma-rule-2` branch (6 commits behind your `master`, no new commits) and tested it on the vanilla dev set as well as copies of the dev set with `ExtraInput=xyz` in the `MISC` column. I got different LAS results, e.g.  67.170 vs 67.179. I repeated the prediction and evaluation and got the same scores as in the first run, so this is not explained by non-deterministic behaviour of parallel computation.

In the baseline, without the new code of PR #18, custom key-value pairs in the `MISC` column should have no effect on the parser as the parser cannot know whether the information provided is input or output and whether it is consistently provided. Only information documented in the UD guidelines should be used, if at all, e.g. `SpaceAfter` information.

**Comment by foxik** [comment]
The code does not use MISC column in any way except for storing them, decoding them to an integer, including them to batches and writing on the output. But the IDs in the batches are not used further.

I thought a bit about whether random number generation could be influenced with non-empty MISC fields but I do not think so. You can compare the training batches generated by the two CoNLL-U files -- they should be the same (except for the unused factor MISC=8), and if they are the same, I believe it is just a bad luck that the results were different two times.

**Comment by jowagner** [comment]
I found the source of differences: ELMo (elmoformanylangs to be precise). I did 3 repetitions with 4 test files that only differ in the MISC column. I tried 2 settings: (a) re-using the same `elmo.npz` file for all predictions, (b) re-calculating the elmo output for each prediction. With the same `elmo.npz` all 12 predictions are the same as expected. With new elmo files for each prediction, I get 12 different predictions.
